### PR TITLE
Add cute_headers lib

### PR DESCRIPTION
**Pull request by mathbunnyru**
Specify library name and version:  **cute_headers/cci.20201022**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/cute_headers/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+class CuteHeadersConan(ConanFile):
+    name = "cute_headers"
+    description = "Various single-file cross-platform C/C++ headers implementing self-contained libraries."
+    topics = ("conan", "various", "pure-c")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/RandyGaul/cute_headers"
+    license = "Unlicense"  # This library (cute-headers) is public domain software, i.e. not copyrighted
+    no_copy_source = True
+
+    def _extract_license(self):
```
Comment (mathbunnyru) [inline recipes/cute_headers/all/conanfile.py:15]: This is how vcpkg extracts license as well.

https://github.com/microsoft/vcpkg/blob/master/ports/cute-headers/portfile.cmake

**Comment by mathbunnyru** [comment]
cute_headers has 2763 stars so far, so it's quite popular, I think. And conan has it in the list https://github.com/conan-io/conan-center-index/issues/621.
And it's easy to package, so I decided to do it :)

File: recipes/cute_headers/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        self.copy(
+            pattern="*.h",
+            dst="include",
+            src=self._source_subfolder,
+            excludes=("examples_cute_*", "test_cute_*")
+        )
+        license_folder = os.path.join(self.package_folder, "licenses")
+        tools.mkdir(license_folder)
+        with open(os.path.join(license_folder, "LICENSE"), 'w') as f:
+            f.write(self._extract_license())
```
Comment (madebr) [inline recipes/cute_headers/all/conanfile.py:40]:
```suggestion
        tools.save(os.path.join(self.package_folder, "licenses", "LICENSE"), self._extract_license())
```

File: recipes/cute_headers/all/test_package/test_package.c [hunk 3]
```diff
@@ -0,0 +1,28 @@
+#ifdef _WIN32
+#include <Windows.h>
```
Comment (madebr) [inline recipes/cute_headers/all/test_package/test_package.c:2]: mingw :smile:
```suggestion
#include <windows.h>
```

File: recipes/cute_headers/all/conanfile.py (hunk 2 above)
Comment (mathbunnyru) [reply recipes/cute_headers/all/conanfile.py]: Nice, thanks :)

File: recipes/cute_headers/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
+from conans import ConanFile, tools
+import os
+import glob
+
+
+class CuteHeadersConan(ConanFile):
+    name = "cute_headers"
```
Comment (madebr) [inline recipes/cute_headers/all/conanfile.py:7]: Do we want it to be `cute_headers` or `cute-headers`?
The name of the github project is `cute_headers`.
vcpkg has named it `cute-headers`.

I vote for `cute-headers`.

Comment (mathbunnyru) [reply recipes/cute_headers/all/conanfile.py:7]: I named it `cute_headers` on purpose.

First of all `def source(self):` is easier to implement when the name is the same as the name of github project.
Second, it is always cute_something in the repo, almost all the files.
And the last reason - there's no `cute-headers` anywhere except vcpkg. If I google `"cute-headers"` (with quotes), it won't give me as much good information as it gives me when I google `"cute_headers"`.

Comment (prince-chrismc) [reply recipes/cute_headers/all/conanfile.py:7]: We have done this in the passed, and it matches the [docs](https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#should-reference-names-use---or-_)

> Recipes should stick to the original name of a library as much as possible.

File: recipes/cute_headers/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    description = "Various single-file cross-platform C/C++ headers implementing self-contained libraries."
+    topics = ("conan", "various", "pure-c")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/RandyGaul/cute_headers"
+    license = "Unlicense"  # This library (cute-headers) is public domain software, i.e. not copyrighted
```
Comment (prince-chrismc) [inline recipes/cute_headers/all/conanfile.py:12]:
```suggestion
    license = "Unlicense"
```

File: recipes/cute_headers/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    no_copy_source = True
+
+    def _extract_license(self):
+        file = os.path.join(self.package_folder, "include/cute_math2d.h")
+        with open(file, 'r') as f:
+            file_content = f.read()
```
Comment (prince-chrismc) [inline recipes/cute_headers/all/conanfile.py:18]: https://docs.conan.io/en/latest/reference/tools.html#tools-load

```suggestion
        file_content = tools.load(file)
```

**Review by prince-chrismc — APPROVED** [review]

**Review by madebr — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
That's smart. Instead of packaging each module, cute-header has all them.

**Pull request merged**